File: 1786-count-the-number-of-consistent-strings/1786-count-the-number-of-consistent-strings.cpp
class Solution {
public:
    int countConsistentStrings(string allowed, vector<string>& words) {
        ios::sync_with_stdio(0);
        cin.tie(0);

        int count = 0;

        for (int i = 0; i < words.size(); i++) {
            bool isConsistent = true;
            for (char word: words[i]) {
                if (allowed.find(word) == string::npos) {
                    isConsistent = false;
                    break;
                }
            }
            if (isConsistent) count++;
        }

        return count;
    }
};

File: 1786-count-the-number-of-consistent-strings/1786-count-the-number-of-consistent-strings.java
class Solution {
    public int countConsistentStrings(String allowed, String[] words) {
        int count = 0;
        int[] allows = new int[26];

        for (char allow: allowed.toCharArray()) {
            allows[allow - 'a'] = 1;
        }

        for (String word: words) {
            boolean consistent = true;
            for (char c: word.toCharArray()) {
                if (allows[c - 'a'] != 1) {
                    consistent = false;
                    break;
                }
            }
            if (consistent) count++;
        }

        return count;
    }
}